subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -19,6 +19,20 @@ LOG_MODULE_DECLARE(usbc_stack, CONFIG_USBC_STACK_LOG_LEVEL);
 
 static const struct smf_state pe_states[PE_STATE_COUNT];
 
+/**
+ * @brief Set the ready state for sink or source.
+ */
+static void pe_set_ready_state(const struct device *dev)
+{
+	struct usbc_port_data *data = dev->data;
+
+	if (data->pe->power_role == TC_ROLE_SOURCE) {
+		pe_set_state(dev, PE_SRC_READY);
+	} else {
+		pe_set_state(dev, PE_SNK_READY);
+	}
```

Comment:
Does the sink handling need to be guarded with `if (!IS_ENABLED(CONFIG_USBC_CSM_SOURCE_ONLY))`?

Reply:
no it is not needed. the state is available with both `CONFIG_USBC_CSM_SOURCE_ONLY ` and `CONFIG_USBC_CSM_SINK_ONLY `
adding ifdef checks here would make the code less readable with no real gain.
I have an additional commit which will be a separate PR where i change so that the pd stack can support both sink and source ports at the same time. If you have multiple ports that can be either sink or source it reasonable to have a stack that can support both.